### PR TITLE
Feature/younghyun get best sellers

### PR DESCRIPTION
**Pull request by younghyunlee22**

### CODE REVIEW

**Comment by younghyunlee22** [comment]
다음 할일: 테스트코드 작성
일단 여기까지 리뷰해주세요.

File: src/services/bookService.js [hunk 1]
```diff
@@ -13,6 +13,26 @@ class BookService {
     }
     return book;
   }
+
+  async getBooksByQueryType(queryType, page = 1, pageSize = 20) {
+    if (!queryType) {
+      throw new Error('Invalid query type');
+    }
+
+    page = Number.isInteger(parseInt(page)) && parseInt(page) > 0 ? parseInt(page) : 1;
```
Comment (hunjoonrhee) [inline src/services/bookService.js]: shouldn't it be declared `const page` ?

File: src/services/bookService.js [hunk 2]
```diff
@@ -13,6 +13,26 @@ class BookService {
...
+    pageSize = Number.isInteger(parseInt(pageSize)) && parseInt(pageSize) > 0 ? parseInt(pageSize) : 20;
```
Comment (hunjoonrhee) [inline src/services/bookService.js]: shouldn't it be declared `const pageSize` ?

File: src/services/bookService.js (hunk 1 above)
Comment (younghyunlee22) [reply src/services/bookService.js]: In a function declaration, default values initialize the parameters, so the variables cannot be redeclared with const or let inside the code block.

File: src/services/bookService.js (hunk 2 above)
Comment (younghyunlee22) [reply src/services/bookService.js]: In a function declaration, default values initialize the parameters, so the variables cannot be redeclared with const or let inside the code block.

File: __tests__/services/bookService.test.js [hunk 3]
```diff
@@ -106,4 +106,90 @@ describe('bookService', () => {
...
+        deleted: false,
+      },
+    ];
+
+    Book.findAll.mockResolvedValue(mockBooks);
+
+    const result = await bookService.getBooksByQueryType('queryType1', 1, 20);
+
+    expect(Book.findAll).toHaveBeenCalledWith(
+      expect.objectContaining({
```
Comment (hunjoonrhee) [inline __tests__/services/bookService.test.js]: I think, expect.objectContaining could be saved here.

File: __tests__/services/bookService.test.js [hunk 4]
```diff
@@ -106,4 +106,90 @@ describe('bookService', () => {
...
+
+    await expect(bookService.getBooksByQueryType('', 1, 20)).rejects.toThrow('Invalid query type');
+  });
+
+  it('should return an empty array if no books are found', async () => {
+    Book.findAll.mockResolvedValue([]);
+    const result = await bookService.getBooksByQueryType('queryType1', 1, 20);
+
+    expect(Book.findAll).toHaveBeenCalledWith(
+      expect.objectContaining({
```
Comment (hunjoonrhee) [inline __tests__/services/bookService.test.js]: I think, expect.objectContaining could be saved here.

**Review by Joel-Hwang — APPROVED** [review]
Looks good to me

**Review by hunjoonrhee — APPROVED** [review]

**Pull request merged**